src/PVRIptvData.cpp

```diff
@@ -1482,3 +1492,23 @@ int PVRIptvData::GetChannelId(const char * strChannelName, const char * strStrea
 
   return abs(iId);
 }
+
+const std::string& PVRIptvData::AddHeaderToStreamUrl (const std::string& url, const std::string& header, const std::string& value) const
```

Comment:
The return type should be `std:string` and not `std::string&` as the returned variable will be on the stack. 

The compiler will optimise so the object is only created once anyway and it will act like a returned reference.